README.md

```diff
@@ -1,58 +1,274 @@
+    - This also failed to track the exact versions of dependencies, meaning even if the right version of "tink_macros" was installed, the wrong version of "tink_core" might be installed, etc.
+- The way haxe and haxelib are tightly coupled makes it difficult to maintain, so development has really stalled for several years. (Haxe Shim breaks this dependency by instead using a standardised way to locate information about dependencies, making it much easier to build a tool like lix.)
+- Juraj, who started Lix, was actually the main maintainer of Haxelib for about two years, and resolved that replacing it was a more effective plan than trying to improve it.
+
+### How does this compare to hmm?
+
+Hmm is another tool that aims to improve package management for haxe projects. It does this by storing version information and installation instructions in "hmm.json", allowing you to restore state based on those instructions.
+
+It's a big step forward from Haxelib, but we think Lix is a step forward again.
+
+Reasons you might prefer hmm:
+
+- you do not want to install NodeJS or NPM.
+- you do not want to use Haxe Shim, you'd prefer to use normal Haxe and normal Haxelib.
+- ... possibly something to do with hxcpp?
+
+Reasons you might prefer Lix:
+
+- Hmm install tracks the dependencies you installed directly, but it doesn't track the sub dependencies, so it's possible when you restore a project the sub-dependency versions might change. In this way builds aren't reproducible and your project might still be subject to "software erosion"
+- Hmm still uses Haxelib to run the installs, and Lix is much faster than Haxelib.
+
+### Is this similar to npm / yarn / cargo / $packageManager?
+
+Lix has taken inspiration from each of these package managers, and tries to learn lessons from each. It is not exactly the same in its implementation as any other package manager.
+
+For example, like "npm" Lix will know how to install dependencies from GitHub, which is great for using unreleased development versions.
+
+Like "yarn", Lix will cache the exact versions installed, including the exact versions of all dependencies, the exact commit SHAs for any dependencies loaded from Github, and more, meaning you have a reproducible build.
+
+Unlike either, Lix does not make a local copy of each library inside a folder in your project, preferring instead to keep the source code in a global folder, to save install time and disk space.
+
+### Can I use these tools without installing them globally?
+
+Yes. When you install just skip the "-g" option:
+
+    npm install haxeshim switchx lix.pm
+
+And then you can run each of them with:
+
+    npm run haxe
+    npm run switchx
+    npm run lix
+
+Note: if you're using NodeJS and package.json in your project, it might be worth adding this to your package.json:
+
+    "scripts": {
+        "postinstall": "npm run lix install"
```

Comment:
This should be `lix install`, every "bin" is set in the environment when executing an npm script.